From c3004f235d68aa890660b532b04714a40abafd78 Mon Sep 17 00:00:00 2001
From: SeongJae Park <sj@kernel.org>
Date: Sat, 20 Jul 2024 11:20:36 -0700
Subject: [PATCH] mm/damon/Kconfig: depends kunit tests on KUNIT=y

DAMON kunit tests are static-built, and depends on KUNIT.  Meanwhile,
KUNIT can be built as module.  In the case, build fails since DAMON
kunit tests in kernel tries to use symbol of kunit module, like below.

    mm/damon/tests/core-kunit.h:394:(.text+0x9c5): undefined reference to `kunit_binary_assert_format'

It is caused by commit 06463781fa14 ("mm/damon/Kconfig: select DAMON
modules to test when test config is set") which changed the dependency
of DAMON kunit tests from KUNIT=y to KUNIT.  Restore the dependencies to
KUNIT=y.

Reported-by: kernel test robot <lkp@intel.com>
Closes: https://lore.kernel.org/oe-kbuild-all/202407202132.JxTrdrIU-lkp@intel.com/
Signed-off-by: SeongJae Park <sj@kernel.org>
---
 mm/damon/Kconfig | 8 ++++----
 1 file changed, 4 insertions(+), 4 deletions(-)

diff --git a/mm/damon/Kconfig b/mm/damon/Kconfig
index f62c957d1291..996c8ff67c8d 100644
--- a/mm/damon/Kconfig
+++ b/mm/damon/Kconfig
@@ -14,7 +14,7 @@ config DAMON
 
 config DAMON_KUNIT_TEST
 	bool "Test for damon" if !KUNIT_ALL_TESTS
-	depends on KUNIT
+	depends on KUNIT=y
 	select DAMON
 	default KUNIT_ALL_TESTS
 	help
@@ -43,7 +43,7 @@ config DAMON_PADDR
 
 config DAMON_VADDR_KUNIT_TEST
 	bool "Test for DAMON operations" if !KUNIT_ALL_TESTS
-	depends on KUNIT
+	depends on KUNIT=y
 	select DAMON_VADDR
 	default KUNIT_ALL_TESTS
 	help
@@ -63,7 +63,7 @@ config DAMON_SYSFS
 
 config DAMON_SYSFS_KUNIT_TEST
 	bool "Test for damon debugfs interface" if !KUNIT_ALL_TESTS
-	depends on KUNIT
+	depends on KUNIT=y
 	select DAMON_SYSFS
 	default KUNIT_ALL_TESTS
 	help
@@ -94,7 +94,7 @@ config DAMON_DBGFS
 
 config DAMON_DBGFS_KUNIT_TEST
 	bool "Test for damon debugfs interface" if !KUNIT_ALL_TESTS
-	depends on KUNIT
+	depends on KUNIT=y
 	select DAMON_DBGFS
 	default KUNIT_ALL_TESTS
 	help
-- 
2.39.2

